### DISCUSSION TITLE
Self signed cert as a parameter

### DISCUSSION THREAD

**Discussion by rxya0** [kcctl/kcctl#332] [Q&A]
Hello,
I am using kcctl to connect to a kafka endpoint that has selfsigned cert. Is there a way to add selfsigned cert chain to truststore?

Thanks!

**Comment by gunnarmorling** [comment]
Hey, is this about Kafka itself, or Kafka Connect? kcctl only connects to the latter, so if only your Kafka brokers require specific config, things should just work? Could you clarify this a bit more, do you e.g. get an exception or similar?

**Reply by rxya0** [reply]
Hi Gunnar!

I got the following exception when running kcctl info against https endpoint.

`javax.ws.rs.ProcessingException: RESTEASY004655: Unable to invoke request: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target`

**Reply by dedovicnermin** [reply]
The above would occur in a connect-cluster with HTTPS + Basic access configured.

Potentially helpful context for solution: 
With `curl`, you'd have to pass the `--cacert`, `--cert`, `--key` flags (along with `--user username:username-password`) to get request all the way through and back. Another way would be to pass the insecure flag instead of the three respective cert flags `curl -k --user <>:<> https://connect.dev.site.domain.com/connectors`

**Reply by dedovicnermin** [reply]
Did some digging, seen workaround already solved [here ](https://github.com/kcctl/kcctl/issues/180#issuecomment-1113894993). Worked for me 
```
kcctl -Djavax.net.ssl.trustStore=/Users/ndedovic/IdeaProjects/ps-cfk/ps-cfk-app/workshop/nermin/truststore.jks -Djavax.net.ssl.trustStorePassword=mystorepassword info
URL:               https://nermin:nermin-secret@connect.dev.gke.confluentps.io
Version:           7.4.0-ce
Commit:            aee97a585bd06866
Kafka Cluster ID:  hBmrCVVaSNa0DaFQdvgutw
```

**Reply by xmlking** [reply]
I have the same issue,  I tried `clientConfig` which didn't fix. 
@gunnarmorling  is there a clean way  to support self-signed certs with basic auth?

`.kcctl file`
```
{
    "currentContext" : "dev",
    "dev" : {
        "cluster" : "https://connect.xyz.internal.com",
        "bootstrapServers" : "kafka.internal.com:1234",
        "username" : "admin",
        "password" : "admin",
        "clientConfig" : {
            "ssl.truststore.location" : "~/truststore.jks",
            "ssl.truststore.password" : "mypassword",
            "ssl.keystore.location" : "~/keystore.jks",
            "ssl.keystore.password" : "mypassword",
            "ssl.key.password" : "mypassword"
        }
    }
}
```
